src/web.rs

```diff
@@ -168,8 +168,11 @@ impl WebRequest<SendClientRequest> {
                 response.status()
             ))?);
         }
+        let raw_body = response.body().await?;
+        let body = std::str::from_utf8(&raw_body).unwrap();
```

Comment:
Calling `.unwrap()` is not the best solution here. Could you try to convert and return the error?

Reply:
Good one!

Changed it from `std::str...` to `String...` since the error was already covered in `src/error.rs`

Reply:
On a second thought - we make an extra copy of the response here. It would be more efficient to create a new error type, e.g.

```rust
pub enum Error {
    ...
    #[error("invalid UTF8 string: {0}")]
    FromStrUtf8Error(#[from] std::str::Utf8Error),
    ...
}
```

and use `std::str::from_utf8(&raw_body)?`;

Reply:
ok, sounds better.

`String` and `str` should get their Error sh*t together :P